### PR TITLE
Make chain hash the default with text format and warn with json formats

### PR DESCRIPTION
**Pull request by jackgerrits**

### CODE REVIEW

**Comment by peterychang** [comment]
I guess there's no reason to allow somebody to disable chain hashing on the text parser? Why can't we make it default behavior for json as well?

**Comment by jackgerrits** [comment]
I don't think disabling it makes sense for the text parser because this is essentially an expansion of the format. Making it default for json is a breaking change we need to deprecate first.

File: vowpalwabbit/parse_args.cc [hunk 1]
```diff
@@ -435,7 +435,7 @@ input_options parse_source(vw& all, options_i& options)
       .add(make_option("no_daemon", all.no_daemon)
                .help("Force a loaded daemon or active learning model to accept local input instead of starting in "
                      "daemon mode"))
-      .add(make_option("chain_hash", parsed_options.chain_hash)
+      .add(make_option("chain_hash", parsed_options.chain_hash_json)
                .help("enable chain hash for feature name and string feature value. e.g. {'A': {'B': 'C'}} is hashed as "
```
Comment (lalo) [inline vowpalwabbit/parse_args.cc:439]: should the help message mention this is default and only option for json?

Comment (jackgerrits) [reply vowpalwabbit/parse_args.cc:439]: Good catch thankyou

File: test/unit_test/dsjson_parser_test.cc [hunk 2]
```diff
@@ -266,8 +267,9 @@ BOOST_AUTO_TEST_CASE(parse_dsjson_cats_no_label)
   }
 }
 )";
-  auto vw = VW::initialize("--dsjson -t --cats 4 --min_value=185 --max_value=23959 --bandwidth 1 --no_stdin --quiet",
-      nullptr, false, nullptr, nullptr);
+  auto vw = VW::initialize(
```
Comment (olgavrou) [inline test/unit_test/dsjson_parser_test.cc:270]: I find this confusing, we need to turn chain hash on for json, even if the feature values are numerical?

Comment (jackgerrits) [reply test/unit_test/dsjson_parser_test.cc:270]: The alternative is keeping track of if a text feature was or wasn't seen and warn only if one was seen. Given we are migrating to this behavior it seems simpler to just enforce it on an input type level?

Comment (olgavrou) [reply test/unit_test/dsjson_parser_test.cc:270]: yeah..

**Review by olgavrou — APPROVED** [review]

**Pull request merged**